lib/EarlyFunctionAnalysis/CFGAnalyzer.cpp

```diff
@@ -964,21 +967,29 @@ CallSummarizer::CallSummarizer(llvm::Module *M,
                                Function *PreCallHook,
                                Function *PostCallHook,
                                llvm::Function *RetHook,
-                               GlobalVariable *SPCSV) :
+                               GlobalVariable *SPCSV,
+                               const FunctionSummaryOracle &Oracle) :
   M(M),
   PreCallHook(PreCallHook),
   PostCallHook(PostCallHook),
   RetHook(RetHook),
   SPCSV(SPCSV),
-  RegistersClobberedPool(M, false) {
+  Oracle(Oracle),
+  RegistersClobberedPool(M, false),
+  WeakReadWritePools(M, false) {
   RegistersClobberedPool.addFnAttribute(llvm::Attribute::ReadOnly);
   RegistersClobberedPool.addFnAttribute(llvm::Attribute::NoUnwind);
   RegistersClobberedPool.addFnAttribute(llvm::Attribute::WillReturn);
+
+  WeakReadWritePools.addFnAttribute(llvm::Attribute::ReadOnly);
+  WeakReadWritePools.addFnAttribute(llvm::Attribute::NoUnwind);
+  WeakReadWritePools.addFnAttribute(llvm::Attribute::WillReturn);
```

Comment:
No need for a new pool, the only goal of these functions is to produce an opaque value to use to clobber the CSV. No  other special meaning is attached to them. I'd just rename `RegistersClobberedPool` to `OpaqueValuesPool` or something along those lines.

I think I suggested this on Mattermost too.

Reply:
OK, double check and let's see.

Reply:
I've verified and not it works. Probably it was caused by some other thing and during further changes it has been fixed.